### PR TITLE
Switch from CryptoJS to WebCrypto API

### PR DESCRIPTION
**Pull request by arjunyel**
From [CryptoJS readme](https://github.com/brix/crypto-js):

> Active development of CryptoJS has been discontinued. This library is no longer maintained.

Removed CryptoJS and replaced it with the WebCrypto API. Also added tests to make sure it still works with a polyfill.

### CODE REVIEW

**Comment by sergiodxa** [comment]
That's because tests still run in Node and Node doesn't have WebCrypto API globally available, you have to import it from the `crypto` package.

This is also the reason Remix Utils uses CryptoJS instead of the WebCrypto API, because of Node.

**Comment by arjunyel** [comment]
Should be fully supported in Node 20 which is now LTS :D

File: src/helpers/base64.ts
Comment (sergiodxa) [inline src/helpers/base64.ts:1]: I would rather prefer to not maintain this myself and instead use a package

**Comment by sergiodxa** [comment]
> Should be fully supported in Node 20 which is now LTS :D It's available on globalThis so node LTS users will have no breaking change besides some methods becoming async. Also CryptoJS is no longer supported so could have security issues. I can't speak for other Remix projects but mine in personally on Cloudflare.

Remix apps requires [Node v18](https://github.com/remix-run/remix/blob/43419f6406667afa17076967083943b2bbff8927/templates/remix/package.json#L29), I'm not sure why but requiring but Remix Utils keeps the same minimum required Node version as the Remix version it supports, since Remix Utils v7 requires Remix v2 it therefore should work with Node v18.

**Comment by arjunyel** [comment]
I agree this would be a breaking change for Node 18 users, they can use `--experimental-global-webcrypto` or the polyfill method from the docs. However CryptoJS is not maintained so the transition to WebCrypto has to happen eventually

**Comment by enZane** [comment]
Any news on this? I don't want to open another thread :p
In a couple of months node lts will be v22

**Comment by sergiodxa** [comment]
As long as Remix supports v18 this will not change

**Comment by arjunyel** [comment]
> As long as Remix supports v18 this will not change

Your Remix auth library uses web crypto so curious why different for this library :) Node 18 users are able to use the Web Crypto polyfill so they wouldn't be left out

**Comment by enZane** [comment]
And what about a way to provide a sharable interface, so this honeypot can be integrated with whatver remix supported runtime

Currently, I faced a problem with cloudflare workers because crypto-js is not edge ready:(

If that makes sense I can try to draft a PR for that instead of a more aggressive transition

**Comment by bitofbreeze** [comment]
Everyone could be happy by making an intermediary package that conditionally exports the needed functions from cryptojs when runtime is node, and otherwise web crypto api. Here's [an example](https://github.com/cloudflare/workers-sdk/blob/main/fixtures/isomorphic-random-example/package.json). Will just need to make a thin wrapper API to make the functions from each package have the same signature.

**Comment by arjunyel** [comment]
> Everyone could be happy by making an intermediary package that conditionally exports the needed functions from cryptojs when runtime is node, and otherwise web crypto api. Here's [an example](https://github.com/cloudflare/workers-sdk/blob/main/fixtures/isomorphic-random-example/package.json) where you can see a different file being exported depending on node or workerd runtime. Here are [some other possible values you can specify for the runtime](https://github.com/stripe/stripe-node/blob/master/package.json#L77). Will just need to make a thin wrapper API to make the functions from each package have the same signature.

I don't think that's necessary, instead older node can just use a WebCrypto polyfill

**Comment by bitofbreeze** [comment]
@arjunyel But then you're adding unnecessary bundle size to all runtimes, not just where it's needed. To make things easier, I found [an existing package that seems to do what I described](https://github.com/TopGunBuild/webcrypto) by the way.

**Comment by bitofbreeze** [comment]
@arjunyel Sure but I guess that is @sergiodxa's concern "this will be a breaking change for those apps, that will now need to supply the Node implementation of WebCrypto", so if you would use that dependency instead, I think it would cover the concern.

**Pull request closed without merging**